### PR TITLE
lightMLFlow `0.2.0`

### PR DESCRIPTION
**Pull request by mrkaye97**
Doing some overhauling. Figured since `resolve_foobar` was getting called everywhere but I'd since changed all the defaults from `NULL` to something the API could actually use, we can just set defaults in the functions now and get rid of a bunch of code.

I also added a ton of type assertions to help with API errors that should be 400 but return 500

### CODE REVIEW

File: R/experiments.R [hunk 1]
```diff
@@ -154,26 +148,25 @@ set_experiment_tag <- function(key, value, experiment_id, client) {
 #' active experiment if both `experiment_id` and `name` are unspecified.
 #'
 #' @param experiment_id ID of the experiment.
-#' @param name The experiment name. Only one of `name` or `experiment_id` should be specified.
+#' @param experiment_name The experiment name. Only one of `name` or `experiment_id` should be specified.
 #' @param client An MLFlow client. Defaults to `NULL` and will be auto-generated.
 #'
 #' @export
-get_experiment <- function(experiment_id, name, client) {
+get_experiment <- function(experiment_id = get_active_experiment_id(), experiment_name = NULL, client = mlflow_client()) {
```
Comment (tonyelhabr) [inline R/experiments.R:155]: i like the change to `experiment_name`

Comment (mrkaye97) [reply R/experiments.R:155]: yeah I want to do a lot more of this in different places, but prob a future PR. there's stuff like `name`, `source`, `path` etc everywhere and they all mean different things so it's super confusing

**Comment by tonyelhabr** [comment]
> Doing some overhauling. Figured since `resolve_foobar` was getting called everywhere but I'd since changed all the defaults from `NULL` to something the API could actually use, we can just set defaults in the functions now and get rid of a bunch of code.
> 
> I also added a ton of type assertions to help with API errors that should be 400 but return 500

Pretty big change in philosophy, but I think it's justified. Eliminating all of the `resolve_*`s also makes things a little easier to reason about. For example, in `create_experiment`, there was nothing in the function signature indicating that `name` must be required vs. all the other params being essentially optional. Now a user can see that all other params have default values, implying that they can be left as is without leading to an error.

**Review by tonyelhabr — APPROVED** [review]
hard for me to check everything, but as long as it passed all the tests, lgtm

**Pull request merged**